### PR TITLE
Drag and drop

### PR DESCRIPTION
**Pull request by markmead**
Currently hitting this error due to assets not being compiled.

Works fine if the `coffescript` and `css` are used on the `index` page. If there's a solution to the asset problem then that would be preferred, however is there much downside with it being inline? It's only used on the `index` view.

Screenshot 2019-10-31 at 17 36 35

It's better to use custom `css` and `js` as the current used on Spina is overcomplicated for this, as it also includes nesting the pages which is not something that is needed on projects.

### CODE REVIEW

File: app/models/spina/project.rb [hunk 1]
```diff
@@ -13,5 +13,9 @@ class Project < ActiveRecord::Base
     accepts_nested_attributes_for :image_collection
 
     scope :newest_first, -> { order('completion_date DESC') }
+
+    def self.by_position
```
Comment (dankmitchell) [inline app/models/spina/project.rb]: I think a scope would be more appropriate for this ordering method i.e.

```
scope : by_position, -> { order('position ASC') }
```

File: app/views/spina/admin/projects/index.html.haml [hunk 2]
```diff
@@ -17,7 +17,7 @@
 .well
   - if @projects.any?
     .dd#pages_list
-      %ol.dd-list
+      %ol.dd-list.sortable
```
Comment (dankmitchell) [inline app/views/spina/admin/projects/index.html.haml]:
```suggestion
      %ol.dd-list#pages_list{data: {:"sort-url" => spina.sort_admin_products_url }}
```

File: app/assets/javascripts/spina/projects/admin/sorting.coffee [hunk 3]
```diff
@@ -0,0 +1,26 @@
+set_positions = ->
```
Comment (dankmitchell) [inline app/assets/javascripts/spina/projects/admin/sorting.coffee]: I think you can loose this file completely. The change I have suggested to the list will hook into the pages sorting JS already in Spina.

File: app/models/spina/project.rb (hunk 1 above)
Comment (markmead) [reply app/models/spina/project.rb]: Is what I've done not considered a way of declaring scope? 

But sure, I'll switch it. Makes sense.

File: app/views/spina/admin/projects/index.html.haml (hunk 2 above)
Comment (markmead) [reply app/views/spina/admin/projects/index.html.haml]: This is why I haven't used what Spina uses. It creates nested projects which might not causes any rendering issues (untested) but it is bad UX.

Screenshot 2019-10-31 at 18 57 52

File: app/models/spina/project.rb (hunk 1 above)
Comment (dankmitchell) [reply app/models/spina/project.rb]: A useful blog https://www.justinweiss.com/articles/should-you-use-scopes-or-class-methods/

**Comment by markmead** [comment]
Still need to find out why the assets aren't working.

**Pull request merged**